File: 프로그래머스/Lv.0/181932. 코드 처리하기/코드 처리하기.py
def solution(code):
    answer = ''
    print("".join(code.split("1"))[::2])
    #"".join(code.split("1"))[::2] or "EMPTY"
    mod = 0
    for i, str in enumerate(code, 0):
        if mod:
            if str == "1":
                mod = 0
            else :
                answer += str if i%2 else ''
        else:
            if str == "1":
                mod = 1
            else :
                answer += str if not i%2 else ''
    
    return answer if len(answer) > 0 else 'EMPTY'

File: 프로그래머스/Lv.0/181932. 코드 처리하기/코드 처리하기.java
class Solution {
    public String solution(String code) {
        String answer = "";
        int mode = 0;
        int cnt = 0;
        StringBuilder sb = new StringBuilder();
        
        char[] ch = code.toCharArray();
        
        for(char nm : ch){
            String temp = String.valueOf(nm);
            if(mode == 0){
                if(!temp.equals("1") && cnt % 2 == 0){
                    sb.append(temp);
                }else if(temp.equals("1")){
                    mode = 1;
                }
            }else if(mode == 1){
                if(!temp.equals("1") && cnt % 2 != 0){
                    sb.append(temp);
                }else if(temp.equals("1")){
                    mode = 0;
                }
            }
            cnt++;
        }
        
        if(sb == null || sb.toString().equals("")){
            sb.append("EMPTY");
        }
        
        answer = sb.toString();
        
        return answer;
    }
}